R/render.r

```diff
@@ -72,7 +72,10 @@ render_page <- function(pkg = ".", name, data, path = "", depth = NULL, quiet =
 data_template <- function(pkg = ".", depth = 0L) {
   pkg <- as_pkgdown(pkg)
 
-  authors <- data_authors(pkg)$main %>%
+  authors <- data_authors(
+    pkg,
+    roles = pkg$meta$authors$footer$roles %||% default_roles()
```

Comment:
I'd pull out `roles <- pkg$meta$authors$footer$roles %||% default_roles()` to keep the flow a bit simpler